haystack/utils/export_utils.py

```diff
@@ -174,6 +174,36 @@ def export_answers_to_csv(agg_results: list, output_file):
     df = pd.DataFrame(data)
     df.to_csv(output_file, index=False)
 
+def export_answers_to_excel(agg_results: list, output_file):
+    """
+    Exports answers coming from finder.get_answers() to an Excel file.
+    :param agg_results: A list of predictions coming from finder.get_answers().
+    :param output_file: The name of the output file.
+    :return: None
+    """
+    if isinstance(agg_results, dict):
+        agg_results = [agg_results]
+
+    assert "query" in agg_results[0], f"Wrong format used for {agg_results[0]}"
+    assert "answers" in agg_results[0], f"Wrong format used for {agg_results[0]}"
+
+    data = {}  # type: Dict[str, List[Any]]
+    data["query"] = []
+    data["prediction"] = []
+    data["prediction_rank"] = []
+    data["prediction_context"] = []
+
+    for res in agg_results:
+        for i in range(len(res["answers"])):
+            temp = res["answers"][i]
+            data["query"].append(res["query"])
+            data["prediction"].append(temp.answer)
+            data["prediction_rank"].append(i + 1)
+            data["prediction_context"].append(temp.context)
+
+    df = pd.DataFrame(data)
```

Comment:
This block is copy pasted from `export_answers_to_csv`. Could you instead refactor `export_answers_to_csv` so that it can be used to dump the dataframe into many formats?

Such as (untested):

```python
def export_answers_to_dataframe(agg_results: list, output_file) -> pd.DataFrame:
    """
    Exports answers coming from finder.get_answers() to an Excel file.
    :param agg_results: A list of predictions coming from finder.get_answers().
    :param output_file: The name of the output file.
    :return: None
    """
    if isinstance(agg_results, dict):
        agg_results = [agg_results]

    assert "query" in agg_results[0], f"Wrong format used for {agg_results[0]}"
    assert "answers" in agg_results[0], f"Wrong format used for {agg_results[0]}"

    data = {}  # type: Dict[str, List[Any]]
    data["query"] = []
    data["prediction"] = []
    data["prediction_rank"] = []
    data["prediction_context"] = []

    for res in agg_results:
        for i in range(len(res["answers"])):
            temp = res["answers"][i]
            data["query"].append(res["query"])
            data["prediction"].append(temp.answer)
            data["prediction_rank"].append(i + 1)
            data["prediction_context"].append(temp.context)

    return pd.DataFrame(data)

def export_answers_to_csv(agg_results: list, output_file):
    df = export_answers_to_dataframe(agg_results, output_file)
    df.to_csv(output_file, index=False)

def export_answers_to_excel(agg_results: list, output_file):
    df = export_answers_to_dataframe(agg_results, output_file)
    df.to_excel(output_file, index=False)
```

(don't forget docstrings and tests)